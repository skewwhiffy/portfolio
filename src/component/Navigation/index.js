import React, { useEffect } from 'react'

const Navigation = props => {
  const { choices, currentChoice, setCurrentChoice } = props
  const onSelect = name => {
    setCurrentChoice(choices.filter(it => it.name === name)[0])
  }
  useEffect(() => {
    document.title = currentChoice.displayName
  });

  return (
    <div className="navigation">
      <ul>
        {
          choices.map(it => {
            return (
              <li className={currentChoice.name === it.name ? 'selected' : ''} key={it.name}>
                <span className="selection" onClick={() => onSelect(it.name)}>
                  {it.displayName}
                </span>
              </li>
            )
          })
        }
      </ul>
    </div>
  )
}

export default Navigation