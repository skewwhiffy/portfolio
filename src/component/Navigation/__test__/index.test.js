import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Navigation from '..';
import userEvent from '@testing-library/user-event';

afterEach(cleanup)

const currentChoice = {
  name: 'choice',
  displayName: 'Dummy Choice For Testing'
}
const anotherChoice = {
  name: 'anotherChoice',
  displayName: 'Another Choice For Testing'
}
const choices = [currentChoice, anotherChoice]

describe('Navigation component', () => {
  let setCurrentChoice
  let asFragment
  let getByText

  beforeEach(() => {
    setCurrentChoice = jest.fn();
    ({ asFragment, getByText } = render(
      <Navigation
        choices={choices}
        currentChoice={currentChoice}
        setCurrentChoice={setCurrentChoice}
      />
    ))
  })

  it('matches snapshot DOM node structure', () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls setCurrentChoice when clicking', () => {
    getByText(anotherChoice.displayName).click()

    expect(setCurrentChoice).toHaveBeenCalledWith(anotherChoice)
  })
})