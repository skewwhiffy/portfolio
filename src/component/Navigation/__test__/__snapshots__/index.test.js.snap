// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="navigation"
  >
    <ul>
      <li
        class="selected"
      >
        <span
          class="selection"
        >
          Dummy Choice For Testing
        </span>
      </li>
      <li
        class=""
      >
        <span
          class="selection"
        >
          Another Choice For Testing
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
